File: services/core/app/models/business/business_config.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional

from typing import Optional, Dict, Any

class BusinessConfig(BaseModel):
    id: Optional[str] = Field(alias="_id", default=None)
    business_id: Optional[str] = None
    # General
    business_name: str
    business_phone: Optional[str] = None
    phone_number: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    website: Optional[str] = None
    industry: Optional[str] = None
    
    # Localization
    language: Optional[str] = "en"
    currency: Optional[str] = "USD"
    timezone: str = "UTC"
    
    # Configuration
    default_duration: Optional[int] = 30
    is_configured: Optional[bool] = False
    
    # API Keys
    vapi_api_key: Optional[str] = None
    openai_api_key: Optional[str] = None
    groq_api_key: Optional[str] = None
    elevenlabs_api_key: Optional[str] = None
    
    # Twilio
    twilio_account_sid: Optional[str] = None
    twilio_auth_token: Optional[str] = None
    twilio_phone_number: Optional[str] = None
    twilio_verify_sid: Optional[str] = None

    # Integrations
    airtable_api_key: Optional[str] = None
    google_calendar_connected: Optional[bool] = False
    google_calendar_credentials: Optional[Dict[str, Any]] = None
    hubspot_credentials: Optional[Dict[str, Any]] = None
    slack_credentials: Optional[Dict[str, Any]] = None
    stripe_credentials: Optional[Dict[str, Any]] = None
    whatsapp_config: Optional[Dict[str, Any]] = None
    
    # AI Settings
    system_prompt: Optional[str] = None
    
    # Automations
    automations: Optional[Dict[str, bool]] = Field(default_factory=dict)
    
    # Scheduling
    business_hours: Optional[dict] = {
        "monday": {"start": "09:00", "end": "17:00", "enabled": True},
        "tuesday": {"start": "09:00", "end": "17:00", "enabled": True},
        "wednesday": {"start": "09:00", "end": "17:00", "enabled": True},
        "thursday": {"start": "09:00", "end": "17:00", "enabled": True},
        "friday": {"start": "09:00", "end": "17:00", "enabled": True},
        "saturday": {"start": "10:00", "end": "14:00", "enabled": True},
        "sunday": {"start": "00:00", "end": "00:00", "enabled": False}
    }

    model_config = ConfigDict(
        populate_by_name=True,
        extra="allow",
        json_schema_extra={
            "example": {
                "_id": "ai_settings_001",
                "business_id": "business_123",
                "business_name": "Royal Fade Barbershop",
                "language": "en",
                "timezone": "Africa/Tunis",
                "default_duration": 30
            }
        }
    )

class BusinessConfigUpdate(BaseModel):
    business_name: Optional[str] = None
    language: Optional[str] = None
    timezone: Optional[str] = None
    currency: Optional[str] = None
    default_duration: Optional[int] = None
    phone_number: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    is_configured: Optional[bool] = None
    
    vapi_api_key: Optional[str] = None
    openai_api_key: Optional[str] = None
    groq_api_key: Optional[str] = None
    elevenlabs_api_key: Optional[str] = None
    
    twilio_account_sid: Optional[str] = None
    twilio_auth_token: Optional[str] = None
    twilio_phone_number: Optional[str] = None
    twilio_verify_sid: Optional[str] = None
    
    airtable_api_key: Optional[str] = None
    google_calendar_connected: Optional[bool] = None
    whatsapp_config: Optional[Dict[str, Any]] = None
    
    system_prompt: Optional[str] = None
    automations: Optional[Dict[str, bool]] = None
    business_hours: Optional[dict] = None
    
    class Config:
        extra = "allow"

File: frontend_next/lib/api/business-config.ts
import api from '@/lib/api';

export interface OpeningHours {
    day_of_week: number; // 0=Monday, 6=Sunday
    open_time: string; // "HH:MM"
    close_time: string; // "HH:MM"
    is_open: boolean;
}

export interface DaySchedule {
    start: string;
    end: string;
    enabled: boolean;
}

export interface BusinessHours {
    monday: DaySchedule;
    tuesday: DaySchedule;
    wednesday: DaySchedule;
    thursday: DaySchedule;
    friday: DaySchedule;
    saturday: DaySchedule;
    sunday: DaySchedule;
    [key: string]: DaySchedule;
}

export interface ServiceDefinition {
    name: string;
    duration_minutes: number;
    price: number;
    description?: string;
    is_active?: boolean;
}

export interface AIConfiguration {
    model?: string;
    temperature?: number;
    max_tokens?: number;
    system_prompt?: string;
    voice_enabled?: boolean;
    voice_model?: string;
}

export interface FeatureFlags {
    voice_agent?: boolean;
    email_notifications?: boolean;
    sms_reminders?: boolean;
    online_booking?: boolean;
    [key: string]: boolean | undefined;
}

export interface WhatsAppConfiguration {
    phone_number_id?: string;
    access_token?: string;
    verify_token?: string;
    webhook_url?: string;
}

export interface BusinessConfig {
    business_id: string;
    business_name: string;
    business_email?: string;
    business_phone?: string;
    business_address?: string;
    email?: string;
    phone_number?: string;
    address?: string;
    timezone: string;
    currency?: string;
    business_hours: BusinessHours;
    services: ServiceDefinition[];
    ai_config: AIConfiguration;
    whatsapp_config: WhatsAppConfiguration;
    features_enabled?: FeatureFlags;
    automations?: { [key: string]: boolean };
    created_at?: string;
    updated_at?: string;
    logo_url?: string;
    primary_color?: string;
    secondary_color?: string;
    vapi_api_key?: string;
    twilio_account_sid?: string;
    twilio_auth_token?: string;
    twilio_phone_number?: string;
    is_configured?: boolean;
}

export interface BusinessConfigUpdate {
    business_name?: string;
    business_email?: string;
    business_phone?: string;
    business_address?: string;
    timezone?: string;
    email?: string;
    phone_number?: string;
    address?: string;
    currency?: string;
    business_hours?: BusinessHours;
    services?: ServiceDefinition[];
    ai_config?: AIConfiguration;
    whatsapp_config?: WhatsAppConfiguration;
    features_enabled?: FeatureFlags;
    automations?: { [key: string]: boolean };
    vapi_api_key?: string;
    twilio_account_sid?: string;
    twilio_auth_token?: string;
    twilio_phone_number?: string;
    is_configured?: boolean;
}

export const businessConfigApi = {
    /**
     * Get current business configuration
     */
    getConfig: async (businessId: string = 'default'): Promise<BusinessConfig> => {
        const response = await api.get<BusinessConfig>('/admin/config');
        return response; // ApiClient already returns .data
    },

    /**
     * Update business configuration
     */
    updateConfig: async (
        data: BusinessConfigUpdate,
        businessId: string = 'default'
    ): Promise<BusinessConfig> => {
        const response = await api.put<BusinessConfig>('/admin/config', data);
        return response; // ApiClient already returns .data
    },

    updateFeatureFlags: async (
        features: FeatureFlags,
        businessId: string = 'default'
    ): Promise<BusinessConfig> => {
        // First get current config to merge
        const current = await businessConfigApi.getConfig();
        const updatedFeatures = { ...current.features_enabled, ...features };

        const response = await api.put<BusinessConfig>(
            '/admin/config',
            { automations: updatedFeatures }
        );
        return response; // ApiClient already returns .data
    },

    updateAutomations: async (
        automations: { [key: string]: boolean },
        businessId: string = 'default'
    ): Promise<BusinessConfig> => {
        const response = await api.put<BusinessConfig>(
            '/admin/config',
            { automations }
        );
        return response; // ApiClient already returns .data
    },

    /**
     * Reload configuration (invalidate cache)
     */
    reloadConfig: async (businessId: string = 'default'): Promise<{ message: string }> => {
        return { message: "Config reloaded" };
    },

    /**
     * Get AI prompt configuration
     */
    getAIPrompt: async (businessId: string = 'default'): Promise<{ system_prompt: string }> => {
        const config = await businessConfigApi.getConfig();
        return { system_prompt: config.ai_config?.system_prompt || "" };
    },

    /**
     * Update AI prompt
     */
    updateAIPrompt: async (
        systemPrompt: string,
        businessId: string = 'default'
    ): Promise<BusinessConfig> => {
        const response = await api.put<BusinessConfig>(
            '/admin/config',
            { system_prompt: systemPrompt }
        );
        return response; // ApiClient already returns .data
    },

    /**
     * Update WhatsApp configuration
     */
    updateWhatsAppConfig: async (
        config: WhatsAppConfiguration,
        businessId: string = 'default'
    ): Promise<BusinessConfig> => {
        const response = await api.put<BusinessConfig>('/admin/config', { whatsapp_config: config });
        return response; // ApiClient already returns .data
    },

    /**
     * Get services list
     */
    getServices: async (businessId: string = 'default'): Promise<ServiceDefinition[]> => {
        const response = await api.get<ServiceDefinition[]>('/services/');
        return response; // ApiClient already returns .data
    },
};
